### PR TITLE
Closes #3122 doseon: generalize DOSEON derivation [create website]

### LINKED ISSUE #3122: Update condition for `DOSEON` derivation
I don't think that the condition works in general. As `ex_single` is used, we have `EXSTDTM == EXENDTM`. I.e., `DOSEON` is only populated if `EXSTDTM == ASTDTM`. If time is collected (in AE or EX), it is unlikely that `DOSEON` is populated for any records. Thus I would add the clearance period for every dose.

If it causes issues due to duplicates, we could use `hours(12)` instead of `days(1)` or `ASTDTM < EXENDTM + days(1)` instead of `ASTDTM <= EXENDTM + days(1)`.

_Originally posted by @bundfussr in https://github.com/pharmaverse/admiral/pull/3061#discussion_r3258931551_

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/cran-release/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

To create the website for the content of this PR add "[create website]" to the title of the pull request.

### CODE REVIEW

**Comment by bundfussr** [comment]
Hi @jeffreyad ,

I've updated the `DOSEON` derivation and added some text to describe the pitfalls and options.

Please have a look and let me know what you think.

File: inst/templates/ad_adae.R [hunk 1]
```diff
@@ -25,12 +25,9 @@ ex <- pharmaversesdtm::ex
...
 ex <- convert_blanks_to_na(ex)
 
-# Create single dose dataset for use in date of last dose derivations.
-# If the exposure dataset contains multi-day dosing records (e.g., one record
-# per treatment period rather than one record per dose), use
-# create_single_dose_dataset() to expand them into one record per dose.
-# Whether this step is necessary depends on how dosing data were collected.
-ex_single <- ex %>%
+# Derive dose start and end dates requires for last dose before event and DOSEON
+# The imputation methods should be adjusted to the study needs.
```
Comment (bms63) [inline inst/templates/ad_adae.R]:
```suggestion
# Derive dose start and end dates required for last dose before event and DOSEON
# The imputation methods should be adjusted to the study needs.
```

File: vignettes/occds.Rmd
Comment (bms63) [inline vignettes/occds.Rmd:1]: What about adding DOSEON and DOSEU into the prose?

Image

File: vignettes/occds.Rmd [hunk 2]
```diff
@@ -298,56 +310,38 @@ dataset_vignette(
...
+e.g., a pill, we have `EXSTDTC == EXENDTC`. I.e., without adding a clearance
+duration, the dose will only be considered as active at the exact time of
+administration.
+
+Adding a clearance duration to the end date may result in overlapping dosing
+intervals for some subjects. Therefore the last dosing record before the adverse
+event is selected in the example below. If overlapping is not expected based on
+the study design and data collection, the `order` and the `mode` argument in the
+`derive_vars_joined()` call below can be removed. Then the function will throw
+an error if overlapping dosing intervals are found.
```
Comment (bms63) [inline vignettes/occds.Rmd:328]: okay - colored me surprised.  I did not realize it would do that

**Review by bms63 — APPROVED** [review]
I just had some minor comments.

File: inst/templates/ad_adae.R (hunk 1 above)
Comment (bundfussr) [reply inst/templates/ad_adae.R]: fixed

File: vignettes/occds.Rmd
Comment (bundfussr) [reply vignettes/occds.Rmd:1]: Good idea, updated.

**Comment by bms63** [comment]
@jeffreyad got time to do a quick review?

**Comment by jeffreyad** [comment]
Yes, I think it looks good

**Review by jeffreyad — APPROVED** [review]
